wordpress-mobile/WordPress-iOS
# Comments: Add the new header component

Refs #20568 

This implements a new (SwiftUI!) header component for Comment screens. Currently, **this is only shown in Reader comments** when the feature flag is enabled. Here are some previews:

Before | After
-|-
![header_before_light](https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/d087b3c2-26d9-4201-8ce4-a15055c3f170) | ![header_after_light](https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/9b04ade4-e95c-49dd-83de-b19e64510d20)
![header_before_dark](https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/96969d51-b0fc-41d6-bfb5-01b6f9e6ec3b) | ![header_after_dark](https://github.com/wordpress-mobile/WordPress-iOS/assets/1299411/5151f1cb-2943-4cad-8ca5-e10343872e83)

⚠️ **Note:** The featured image thumbnail is not included yet in this PR. It will be added through a separate PR.

## To test

- Launch the app. 
- Go to App Settings > Debug and ensure that the `Comment Moderation Update` flag is toggled on.
- Go to any Reader comments.
- Verify that the new header is shown.

## Regression Notes
1. Potential unintended areas of impact
Should be none. The feature is hidden behind a feature flag.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes.

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:


UI Changes testing checklist:
